### PR TITLE
remove usage of xcstringstool-generated strings to improve performance #1458

### PR DESCRIPTION
**Pull request by bryanmontz**
## Issues covered
#1458 

## Description
tldr:
Removes most of the usage of generated strings from xcstringstool due to an observable performance decrease.

Backstory:
While researching and testing the app's performance, I noticed that most of the worst performing views in the app had lots of occurrences of generated strings (like `.localized.stringKey`), and when those were replaced with just the key, performance improved significantly. In one case, the `body` method's performance increased 70X.

Although it doesn't cover performance, here's an interesting article comparing `LocalizedStringResource`, `LocalizedStringKey`, and `String(localized:)`:
https://medium.com/@nicmcconn/ios-localization-localizedstringresource-vs-localizedstringkey-vs-string-56cb519cf098

Note that I left a handful of the generated strings because the standard, simple localization method doesn't work in unit tests.

## Screenshots/Video
There are no intended UI changes. However, here's a video of the app running in Chinese to show that localization is working as expected:
localized

### CODE REVIEW

**Comment by joshuatbrown** [comment]
👀

**Review by joshuatbrown — COMMENTED** [review]
I looked at the slowest Nos views by total duration on `main`. There were 22 views that took over 100 µs total duration in my Feed scrolling test (3 swipes). I compared the average duration of each of them to their average duration on this branch, and performance is consistently better on this branch. Sometimes by 30%, sometimes by 50%, and maybe even more. Very rarely performance was 10% worse (for maybe 3 views of 22), but I'm chalking that up to external variables. These performance improvements are great!

I found an issue with some things on the profile screen:
- the known followers view has asterisks instead of bold text
- the following and relays numbers aren't there -- instead, we have text that just says "Following following" and "Relays relays"

Simulator Screenshot - iPhone 16 Pro - 2024-10-15 at 11 45 13

**Comment by pelumy** [comment]
This is a lot of work @bryanmontz  💪

File: Nos/Views/Components/Form/NosTextEditor.swift [hunk 1]
```diff
@@ -3,11 +3,16 @@ import SwiftUINavigation
...
 struct NosTextEditor: View {
     
-    var label: LocalizedStringResource
+    var label: LocalizedStringKey
     @Binding var text: String
     
+    init(_ label: LocalizedStringKey, text: Binding<String>) {
+        self.label = label
+        self._text = text
+    }
```
[the remark this answers is not in the document]
Comment (bryanmontz) [reply Nos/Views/Components/Form/NosTextEditor.swift:12]: I did that in order to match the SwiftUI APIs (`Text`, `Button`, etc.) so that you didn't need the argument label "label:" when using a `LocalizedStringKey`.

That way call sites will look like:
```swift
NosTextEditor("bio")
```

rather than:
```swift
NosTextEditor(label: "bio")
```

**Comment by bryanmontz** [comment]
Thanks for reviews! Sorry for the size.

**Comment by joshuatbrown** [comment]
👀

File: Nos/Views/Relay/RelayDetailView.swift [hunk 2]
```diff
@@ -2,62 +2,61 @@ import SwiftUI
 
 struct RelayDetailView: View {
 
-    var relay: Relay
+    let relay: Relay
```
Comment (joshuatbrown) [inline Nos/Views/Relay/RelayDetailView.swift:5]: :100:

File: Nos/Views/Components/Form/NosTextEditor.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (joshuatbrown) [reply Nos/Views/Components/Form/NosTextEditor.swift:12]: Awesome. Thanks for the explanation.

File: CHANGELOG.md [hunk 3]
```diff
@@ -9,6 +9,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+- Remove most usage of xcstringstool-generated strings to improve performance. [#1458](https://github.com/planetary-social/nos/issues/1458)
```
Comment (joshuatbrown) [inline CHANGELOG.md:12]:
```suggestion
### Release Notes
- Remove most usage of xcstringstool-generated strings to improve performance. [#1458](https://github.com/planetary-social/nos/issues/1458)
```

**Review by joshuatbrown — COMMENTED** [review]
Looking good. Just a couple more things, then I'm good with this. Tested on the simulator again and everything was perfect there.

File: Nos/Views/Components/Button/ActionButton.swift [hunk 4]
```diff
@@ -31,6 +31,76 @@ struct ActionButton: View {
...
+        textColor: SwiftUI.Color = Color.white,
+        depthEffectColor: Color = Color.actionPrimaryDepthEffect,
+        backgroundGradient: LinearGradient = LinearGradient(
+            colors: [
+                Color.actionPrimaryGradientTop,
+                Color.actionPrimaryGradientBottom
+            ],
+            startPoint: .bottomLeading,
+            endPoint: .topTrailing
+        ),
```
[the remark this answers is not in the document]
Comment (bryanmontz) [reply Nos/Views/Components/Button/ActionButton.swift:49]: Love it. Thanks. I didn't know that extension existed.

**Comment by joshuatbrown** [comment]
👀

**Comment by joshuatbrown** [comment]
👀

**Review by joshuatbrown — APPROVED** [review]
This all looks great! The issues I found before are all fixed! Awesome!
